tests/lib/rules/jsx-sort-props.js

```diff
@@ -494,6 +608,170 @@ ruleTester.run('jsx-sort-props', rule, {
+          b={{
+            bB: 1,
+          }}
+          c
+        />
+      `,
+      options: multilineAndShorthandFirstArgs,
+      errors: [expectedMultilineFirstError, expectedShorthandFirstError],
+      output: `
+        <App
+          c
+          b={{
+            bB: 1,
+          }}
+          a={1}
+        />
+      `,
+    },
+    // multiline last
+    {
+      code: `
+        <App
+          a={{
+            aA: 1,
+          }}
+          b
+        />
+      `,
+      options: multilineLastArgs,
+      errors: [expectedMultilineLastError],
+      output: `
+        <App
+          b
+          a={{
+            aA: 1,
+          }}
+        />
+      `,
+    },
+    {
+      code: `
+        <App
+          a={{
+            aA: 1,
+          }}
+          b
+          onClick={() => ({
+            c: 1
+          })}
+          d="dD"
+          e={() => ({
+            eE: 1
+          })}
+          f
+        />
+      `,
+      options: multilineAndShorthandAndCallbackLastArgs,
+      errors: 3,
+      output: `
+        <App
+          d="dD"
+          a={{
+            aA: 1,
+          }}
+          e={() => ({
+            eE: 1
+          })}
+          b
+          f
+          onClick={() => ({
+            c: 1
+          })}
+        />
+      `,
+    },
+    {
+      code: `
+        <Typography
+          float
+          className={classNames(classes.inputWidth, {
+            [classes.noBorder]: isActive === "values",
+          })}
+          disabled={isDisabled}
+          initialValue={computePercentage(number, count)}
+          InputProps={{
+            ...customInputProps,
+          }}
+          key={index}
+          isRequired
+          {...sharedTypographyProps}
+          ref={textRef}
+          min="0"
+          name="fieldName"
+          placeholder={getTranslation("field")}
+          onValidate={validate}
+          inputProps={{
+            className: inputClassName,
+          }}
+          outlined
+          {...rest}
+        />
+      `,
+      options: [
+        {
+          multilineLast: true,
+          shorthandFirst: true,
+          callbacksLast: true,
+          reservedFirst: true,
+          ignoreCase: true,
+        },
+      ],
+      output: `
+        <Typography
+          key={index}
+          float
+          isRequired
+          disabled={isDisabled}
+          initialValue={computePercentage(number, count)}
+          className={classNames(classes.inputWidth, {
+            [classes.noBorder]: isActive === "values",
+          })}
+          InputProps={{
+            ...customInputProps,
+          }}
+          {...sharedTypographyProps}
+          ref={textRef}
+          outlined
+          min="0"
+          name="fieldName"
+          placeholder={getTranslation("field")}
+          inputProps={{
+            className: inputClassName,
+          }}
+          onValidate={validate}
+          {...rest}
+        />
+      `,
+      errors: 6,
```

Comment:
let's use an array of objects with messageId and line number, so we can make sure the right lines are being warned.